log/log.go

```diff
@@ -0,0 +1,151 @@
+package log
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"log"
+	"sync"
+	"time"
+)
+
+// Singleton LogSet
+var loggerSet = NewLoggerSet(Info)
+
+// A public reference to the singleton LogSet if required for injection
+var L = loggerSet
+
+// A simple logging interface
+type Logger interface {
+	// Log logs a message with the specified log level and error (if applicable), along with the provided context.
+	//
+	// level: The log level for the message.
+	// message: The message to log.
+	// err: An optional error associated with the message (if any).
+	// ctx: An optional context to log with the message.
+	Log(level LogLevel, message string, err error, ctx context.Context)
+
+	// SetMinimumLevel sets the minimum log level; logs below this level will be ignored
+	SetMinimumLevel(level LogLevel)
+
+	// If applicable to the logger type, closes the logger with a timeout to flush all buffered log entries before returning
+	Close(timeout time.Duration) error
+}
+
+// A set of log sinks (e.g., stdout, file, Sentry, etc.)
+// Logs are sent to all sinks
+type LoggerSet struct {
+	loggers []Logger
+}
+
+// Create a new log set, with the standard logger
+func NewLoggerSet(minimumLevel LogLevel) *LoggerSet {
+	logSet := &LoggerSet{}
+	logSet.AddLogger(NewStandardLogger(minimumLevel))
+
+	return logSet
+}
+
+func (l *LoggerSet) AddLogger(logger Logger) {
+	_addLogger(l, logger)
+}
+
+func AddLogger(logger Logger) {
```

Comment:
It would be nice if these different things were in different files.  The LoggerSet implementation in a `logger_set.go`, and the global instance and functions in a different file.